paint/data/stac_client.py

```diff
@@ -0,0 +1,560 @@
+            log.info("Loading catalogs for desired heliostats. ")
+            heliostat_catalogs_list = [
+                self.get_catalog(
+                    href=mappings.HELIOSTAT_CATALOG_URL % (heliostat, heliostat)
+                )
+                for heliostat in heliostats
+            ]
+
+        # Log warning if now collection keys provided.
+        if collections is None:
+            log.warning(
+                "No collections selected - downloading data for all collections!"
+            )
+            get_calibration = True
+            get_deflectometry = True
+            get_properties = True
+        # Check if collection keys provided are acceptable.
+        else:
+            allowed_values = {
+                mappings.SAVE_DEFLECTOMETRY.lower(),
+                mappings.SAVE_CALIBRATION.lower(),
+                mappings.SAVE_PROPERTIES.lower(),
+            }
+            for item in collections:
+                if item not in allowed_values:
+                    raise ValueError(
+                        f"The heliostat collection must be one of: `deflectometry, calibration, "
+                        f"properties`! The key {item} is not accepted!"
+                    )
+            # Set boolean flags based on the presence of each allowed value in collections.
+            get_calibration = mappings.SAVE_CALIBRATION.lower() in collections
+            get_deflectometry = mappings.SAVE_DEFLECTOMETRY.lower() in collections
+            get_properties = mappings.SAVE_PROPERTIES.lower() in collections
+
+        # Log warning if now filtered calibration keys provided.
+        if get_calibration and filtered_calibration_keys is None:
+            log.warning(
+                "No calibration filters provided - downloading all calibration data, i.e. raw, cropped, flux, centered"
+                "flux and calibration properties!"
+            )
+        # Check if keys provided to the filtered_calibration_key dictionary are acceptable
+        elif get_calibration and filtered_calibration_keys is not None:
+            accepted_keys = [
+                mappings.CALIBRATION_RAW_IMAGE_KEY,
+                mappings.CALIBRATION_CROPPED_IMAGE_KEY,
+                mappings.CALIBRATION_FLUX_IMAGE_KEY,
+                mappings.CALIBRATION_FLUX_CENTERED_IMAGE_KEY,
+                mappings.CALIBRATION_PROPERTIES_KEY,
+            ]
+            for key in filtered_calibration_keys:
+                if key not in accepted_keys:
+                    raise ValueError(
+                        f"The filtered calibration keys can only be one or more of: `raw_image, cropped_image, "
+                        f"flux_image, flux_centered_image, calibration_properties'! The key {key} is not accepted!"
+                    )
+
+        # Error if only a start or end date is provided, but not both.
+        if (start_date and not end_date) or (end_date and not start_date):
```

Comment:
Not sure whether that would be necessary but should be also check for `start_date < end_date`?

Reply:
Done :)